projects/storefrontlib/cms-components/misc/icon/icon.component.ts

```diff
@@ -47,87 +47,100 @@ export class IconComponent {
    * The cxIcon directive is bound to the icon type. You can feed the `ICON_TYPE` to
    * accomplish a configurable button in the UI.
    */
-  @Input() set cxIcon(type: ICON_TYPE) {
+  @Input()
+  public set cxIcon(type: ICON_TYPE_STRING) {
     this.setIcon(type);
   }
 
   /**
    * The type input parameter is bound to the icon type. You can feed the `ICON_TYPE` to
    * accomplish a configurable button in the UI.
    */
-  @Input() set type(type: ICON_TYPE) {
+  @Input()
+  public set type(type: ICON_TYPE_STRING) {
+    // XXX this clashes with the type attribute of some html form elements, e.g. input or button
     this.setIcon(type);
   }
 
-  /**
-   * the icon provides an html fragment that is used to add SVG or text based icons.
-   */
-  icon: SafeHtml | undefined;
-
-  /**
-   * The `flip-at-rtl` class is added to the DOM for the style layer to flip the icon in RTL direction.
-   */
-  @HostBinding('class.flip-at-rtl') flipAtRtl: boolean;
-
-  /**
-   * The `flip-at-ltr` class is added to the DOM for the style layer to flip the icon in LTR direction.
-   */
-  @HostBinding('class.flip-at-ltr') flipAtLtr: boolean;
-
   /**
    * Maintains the applied style classes so we can remove them when the
    * icon type changes at run time.
    */
-  protected styleClasses: string[];
+  @HostBinding('class')
+  public styleClasses: string[] = [];
+
+  private iconValue?: string;
+  private iconResourceType: IconResourceType;
 
   constructor(
     protected iconLoader: IconLoaderService,
-    protected elementRef: ElementRef<HTMLElement>,
-    protected renderer: Renderer2
+    protected sanitizer: DomSanitizer
   ) {}
```

Comment:
Removing a constructor parameter is a breaking change.

Reply:
Same for adding a non-optional constructor parameter. I'll definitely have to fix this.